### PR TITLE
feat(service-desk): UI for companies and procurers

### PR DESCRIPTION
**Pull request by GunnlaugurG**
## What

Added UI for looking up procurers for companies by nationalId or name
Added field resolver to fetch company relations and backend logic to support UI

## Why

So authenticated users with specific scope can look up procurers for companies

## Screenshots / Gifs

Nothing found
image

Listi af fyrirtækjum
image
yrirtækjum

Prókúruhafatengsl
image

No relationships found
image

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/portals/admin/service-desk/src/screens/Procurers/Procurers.tsx
Comment (snaerseljan) [inline libs/portals/admin/service-desk/src/screens/Procurers/Procurers.tsx:26]: It would be great to create a back button component under `libs/portals/admin/core/src/components/BackButton`

And reuse here and in ids admin

File: libs/portals/admin/service-desk/src/screens/Procurers/Procurers.tsx
Comment (snaerseljan) [inline libs/portals/admin/service-desk/src/screens/Procurers/Procurers.tsx:47]: The Grid and GridRow makes the cards overflow the container. Just use flex and flex gap.
Something like 

```
 <Box display="flex" flexDirection="column" rowGap={2}>
            {company.procurers?.map((procurer) => (
              <Box
                key={`procure-${procurer.nationalId}`}
                display={'flex'}
                borderRadius={'large'}
                border={'standard'}
                width={'full'}
                paddingX={4}
                paddingY={3}
                justifyContent={'spaceBetween'}
                alignItems={'center'}
              >
                <Stack space={1}>
                  <Text variant="h3">{procurer.name}</Text>
                  <Text variant={'default'}>
                    {formatNationalId(procurer.nationalId)}
                  </Text>
                </Stack>
              </Box>
            ))}
          </Box>
```

And remember to extract to a CardComponent like mentioned above

File: libs/portals/admin/service-desk/src/utils/formatNationalid.ts
Comment (snaerseljan) [inline libs/portals/admin/service-desk/src/utils/formatNationalid.ts:3]: Has this not been done before somewhere? Can it be reused?

File: libs/portals/admin/service-desk/src/utils/formatNationalid.ts
Comment (snaerseljan) [inline libs/portals/admin/service-desk/src/utils/formatNationalid.ts:4]: You can skip the else here it is redundant

File: libs/portals/admin/service-desk/src/hooks/useNationalId.tsx
Comment (snaerseljan) [inline libs/portals/admin/service-desk/src/hooks/useNationalId.tsx:16]: You created a formatNationalId util method that you should rather use.

File: libs/portals/admin/service-desk/src/hooks/useNationalId.tsx
Comment (snaerseljan) [inline libs/portals/admin/service-desk/src/hooks/useNationalId.tsx:10]: After getting better context I see that this hook is redundant. You could just use the utility method and remove this hook. Unless you where planning of using it?

**Review by snaerseljan — DISMISSED** [review]
Nice work Gulli. Really like the overall structure. 

Most of the comments are nits but others are relevant.

I would personally like to have two nice to haves done. Well at least one of the, that is to have an empty state. I have requested design to implement.
Then it would be really nice to see the SSN to be formatted on the fly in the search box. But that is maybe more of a nice to have.

Other than that great job! 👍

File: libs/portals/admin/service-desk/src/screens/Companies/Companies.tsx
Comment (snaerseljan) [reply libs/portals/admin/service-desk/src/screens/Companies/Companies.tsx:45]: See example in a comment below

File: libs/portals/admin/service-desk/src/hooks/useNationalId.tsx
Comment (GunnlaugurG) [reply libs/portals/admin/service-desk/src/hooks/useNationalId.tsx:16]: Was going to delete this after I created the formatNationalId, must have forgotten :D

File: handbook/technical-overview/api-design-guide/documentation.md
Comment (eirikurn) [inline handbook/technical-overview/api-design-guide/documentation.md:154]: Not relevant replace.

File: libs/api/domains/company-registry/src/lib/resolvers/rskCompanyInfo.resolver.ts
Comment (eirikurn) [inline libs/api/domains/company-registry/src/lib/resolvers/rskCompanyInfo.resolver.ts:19]: We need to test and make sure that this field resolver only works if you have the serviceDesk scope.

File: libs/auth/scopes/src/lib/api.scope.ts
Comment (eirikurn) [inline libs/auth/scopes/src/lib/api.scope.ts:18]: This should be in the admin scope list.

File: libs/clients/national-registry/v2/README.md
Comment (eirikurn) [inline libs/clients/national-registry/v2/README.md:60]: Not valid replace?

File: libs/clients/rsk/procuring/project.json
Comment (eirikurn) [inline libs/clients/rsk/procuring/project.json:24]: Can you not use the `getOpenAPI` endpoint to get the client config?

File: libs/clients/rsk/procuring/src/lib/RskProcuringClient.ts
Comment (eirikurn) [inline libs/clients/rsk/procuring/src/lib/RskProcuringClient.ts:13]: Should rename this lib and services to "relationships"

File: libs/auth-api-lib/src/lib/delegations/delegations.service.ts
Comment (eirikurn) [inline libs/auth-api-lib/src/lib/delegations/delegations.service.ts:445]: Need to make sure that these endpoints work without having the admin scope. Should be enough to test that procuration holder delegations still work after merging to main.

**Review by Thorrik58 — APPROVED** [review]
LGTM on codeowned files.

**Review by thordurhhh — APPROVED** [review]
Code owned files looking good.

File: libs/api/domains/company-registry/src/lib/resolvers/rskCompanyInfo.resolver.ts
Comment (eirikurn) [inline libs/api/domains/company-registry/src/lib/resolvers/rskCompanyInfo.resolver.ts:33]: 🙌

File: libs/clients/rsk/relationships/README.md
Comment (eirikurn) [inline libs/clients/rsk/relationships/README.md:1]:
```suggestion
<!-- gitbook-navigation: "Relationships" -->
```

File: libs/clients/rsk/relationships/README.md
Comment (eirikurn) [inline libs/clients/rsk/relationships/README.md:16]:
```suggestion
| `getLegalEntityRelationships()` | `@rsk.is/prokura:admin` | Returns relationships for individuals related to a specific legal entity. The data is sensitive and should only be accessible to Ísland.is service desk admins |
| `getIndividualRelationships()`  | `@rsk.is/prokura` | Returns relationships for companies related to a specific individual                                                                                                        |
```

File: libs/clients/rsk/relationships/README.md
Comment (eirikurn) [inline libs/clients/rsk/relationships/README.md:11]:
```suggestion
Following table describes methods and which scopes are needed to access them, e.g. by the IDS machine client configured in the service using this client.
```

**Review by eirikurn — APPROVED** [review]
Approved with a few suggestions 👍

**Review by kksteini — APPROVED** [review]
CODEOWNER files LGTM

**Review by lodmfjord — APPROVED** [review]
LGTM

**Pull request merged**
